marker/v2/processors/text.py

```diff
@@ -0,0 +1,33 @@
+from marker.v2.processors import BaseProcessor
+from marker.v2.schema import BlockTypes
+from marker.v2.schema.document import Document
+from marker.v2.schema.text.line import Line
+from typing import List
+
+
+class TextProcessor(BaseProcessor):
+    block_types = (BlockTypes.Text, BlockTypes.TextInlineMath)
+    min_continuation_width_ratio = 0.98
+
+    def __init__(self, config):
+        super().__init__(config)
+
+    def __call__(self, document: Document):
+        for page in document.pages:
+            for block in page.children:
+                if block.block_type in self.block_types:
+                    if block.structure is None:
+                        continue
+
+                    if not len(block.structure) > 3:  # Skip non paragraphs
+                        continue
+
+                    next_block = document.get_next_block(block)
+                    if next_block and next_block.block_type not in self.block_types:
+                        continue
+
+                    lines: List[Line] = [page.get_block(block_id) for block_id in block.structure]
+                    avg_width = sum([l.polygon.width for l in lines]) / len(lines)
+
+                    if lines[-1].polygon.width >= avg_width * self.min_continuation_width_ratio:
```

Comment:
Doesn't this lead to a lot of false positives?  Basically, if the last line is close to the average line width, then the next line is a continuation?

I think there are some other options to do this:

- Look explictly for x position, and see if the line runs to the max end x position of all the lines
- Check if there is a hyphen, which is an explicit continuation
- Check if it's the last text block in the column or on the page

I think the best option is to see if it's the last block in the column or on the page, then verify the x position of the current line equals max x position of all lines in the current block.  Then you also have to look at the next block to make sure the first line isn't indented.  Basically three criteria versus one.

Reply:
Done!